interpreter/src/main/scala/pl/touk/nussknacker/engine/spel/Typer.scala

```diff
@@ -564,6 +566,22 @@ private[spel] class Typer(commonSupertypeFinder: CommonSupertypeFinder,
 }
 
 object Typer {
+  def default(classLoader: ClassLoader, expressionConfig: ExpressionDefinition[_], dictRegistry: DictRegistry, typeDefinitionSet: TypeDefinitionSet): Typer = {
+    val evaluationContextPreparer = EvaluationContextPreparer.default(classLoader, expressionConfig)
+
+    val strictTypeChecking = expressionConfig.strictTypeChecking
+    val classResolutionStrategy = if (strictTypeChecking) SupertypeClassResolutionStrategy.Intersection else SupertypeClassResolutionStrategy.Union
+    val commonSupertypeFinder = new CommonSupertypeFinder(classResolutionStrategy, strictTypeChecking)
+    new Typer(commonSupertypeFinder,
+      new KeysDictTyper(dictRegistry),
```

Comment:
We should pass `SpelDictTyper` instead of `DictRegistry` into `default` method. `SpelExpressionSuggester` should create `Typer` using `LabelsDictTyper` instead of `KeysDictTyper`. I know it is a little overcomplicated; I'll likely find the energy to simplify it someday. In short, it should be done this way because during suggestions you work on labels, not keys (ids).

Reply:
Ok, looks like you know what you're talking about 😅 Changed that.